Samples/HelloWorld/OrleansClient/Program.cs

```diff
@@ -16,8 +16,27 @@ public class Program
         static int Main(string[] args)
         {
             var config = ClientConfiguration.LocalhostSilo();
-            const int initializeAttemptsBeforeFailing = 5;
-            int attemp = 0;
+            try
+            {
+                InitializeWithRetries(config, initializeAttemptsBeforeFailing: 5);
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Orleans client initialization failed failed due to {ex}");
+
+                Console.ReadLine();
+                return 1;
+            }
+
+            DoClientWork().Wait();
+            Console.WriteLine("Press Enter to terminate...");
+            Console.ReadLine();
+            return 0;
+        }
+
+        private static void InitializeWithRetries(ClientConfiguration config, int initializeAttemptsBeforeFailing)
+        {
+            int attempt = 0;
             while (true)
             {
```

Comment:
maybe we shouldn't use wild loop here, it would go infinite if the exception we caught it not SiloUnavailableException. We can do something like this. 
```
while(attempt < initializeAttemptsBeforeFailing)
{
  try {
       GrainClient.Init;
       ...
     break;
  }catch (Exception e)
 {
  attempt ++;
  Console.writeLine($"Attempt {attempt} of {initializeAttemptsBeforeFailing} failed due to {e}"); 
 }
Thread.Sleep(TimeSpan.FromSeconds(2));
```
 
}